packages/hierarchy-builder/src/test/internal/operators/grouping/PropertiesGrouping.test.ts

```diff
@@ -0,0 +1,1482 @@
+                },
+              },
+            },
+          }),
+          createTestProcessedInstanceNode({
+            key: { type: "instances", instanceKeys: [{ className: "TestSchema.Class", id: "0x2" }] },
+            processingParams: {
+              grouping: {
+                byProperties: {
+                  fullClassName: "TestSchema.Class",
+                  propertyGroups: [{ propertyName: "PropertyName", propertyValue: 5, ranges: [{ fromValue: 1, toValue: 4, rangeLabel: "rangeLabel" }] }],
+                },
+              },
+            },
+          }),
+        ];
+        const ecClass = { fullName: "TestSchema.Class" } as unknown as ECClass;
+        classStubs.stubEntityClass({
+          schemaName: "TestSchema",
+          className: "Class",
+          is: async () => true,
+        });
+        const propertyInfo: propertiesGrouping.PropertyGroupInfo = {
+          ecClass,
+          previousPropertiesGroupingInfo: [],
+          propertyGroup: { propertyName: "PropertyName", ranges: [{ fromValue: 1, toValue: 5, rangeLabel: "rangeLabel" }] },
+        };
+        const expectedGroupingNodeKey: GroupingNodeKey = {
+          type: "property-grouping:range",
+          propertyName: "PropertyName",
+          fullClassName: "TestSchema.Class",
+          fromValue: 1,
+          toValue: 5,
+        };
+        expect(await propertiesGrouping.createPropertyGroups(metadataProvider, nodes, propertyInfo, formatter)).to.deep.eq({
+          groupingType: "property",
+          grouped: [
+            createTestProcessedGroupingNode({
+              label: "rangeLabel",
+              key: expectedGroupingNodeKey,
+              children: [{ ...nodes[0], parentKeys: [...nodes[0].parentKeys, expectedGroupingNodeKey] }],
+            }),
+          ],
+          ungrouped: [nodes[1]],
+        });
+      });
+
+      it("groups multiple nodes, when property values fit into range", async () => {
+        const nodes = [
+          createTestProcessedInstanceNode({
+            key: { type: "instances", instanceKeys: [{ className: "TestSchema.Class", id: "0x1" }] },
+            processingParams: {
+              grouping: {
+                byProperties: {
+                  fullClassName: "TestSchema.Class",
+                  propertyGroups: [{ propertyName: "PropertyName", propertyValue: 5, ranges: [{ fromValue: 1, toValue: 5, rangeLabel: "rangeLabel" }] }],
+                },
+              },
+            },
+          }),
+          createTestProcessedInstanceNode({
+            key: { type: "instances", instanceKeys: [{ className: "TestSchema.Class", id: "0x2" }] },
+            processingParams: {
+              grouping: {
+                byProperties: {
+                  fullClassName: "TestSchema.Class",
+                  propertyGroups: [{ propertyName: "PropertyName", propertyValue: 5, ranges: [{ fromValue: 1, toValue: 5, rangeLabel: "rangeLabel" }] }],
```

Comment:
maybe change the property value to fit into the range, but not be equal to other node's value

Reply:
Changed.